source/slang/ir-glsl-legalize.cpp

```diff
@@ -285,80 +302,121 @@ GLSLSystemValueInfo* getGLSLSystemValueInfo(
         // TODO: deal with `gl_SampleMaskIn` when used as an input.
 
         // TODO: type conversion is required here.
+
+        // uint in hlsl, int in glsl
+        // https://www.opengl.org/sdk/docs/manglsl/docbook4/xhtml/gl_SampleMask.xml
+
+        requiredType = builder->getBasicType(BaseType::Int);
+
         name = "gl_SampleMask";
     }
     else if(semanticName == "sv_depth")
     {
+        // Float in hlsl & glsl
+        // https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/gl_FragDepth.xhtml
         name = "gl_FragDepth";
     }
     else if(semanticName == "sv_depthgreaterequal")
     {
         // TODO: layout(depth_greater) out float gl_FragDepth;
+
+        // Type is 'unknown' in hlsl 
         name = "gl_FragDepth";
     }
     else if(semanticName == "sv_depthlessequal")
     {
         // TODO: layout(depth_greater) out float gl_FragDepth;
+
+        // 'unknown' in hlsl, float in glsl 
         name = "gl_FragDepth";
     }
     else if(semanticName == "sv_dispatchthreadid")
     {
+        // uint3 in hlsl, uvec3 in glsl
```

Comment:
An important point that these comments leave out is that while there might be an "official" type for most of these in HLSL, in practice the user is allowed to declare almost anything that the HLSL compiler can implicitly convert to/from the correct type.

Reply:
So by this argument - does that imply we should always specify the glsl type, such that the type convert will be attempted for glsl output, and in doing so allowing more valid hlsl work with glsl output?